variadics/src/lib.rs

```diff
@@ -198,6 +223,14 @@ where
         let (item, rest) = self;
         rest.reverse().extend((item, ()))
     }
+    fn reverse_ref(this: Self::AsRefVar<'_>) -> <Self::Reverse as VariadicExt>::AsRefVar<'_> {
+        let (item, rest) = this;
+        let out = Rest::reverse_ref(rest).extend((item, ()));
+        // TODO!!!
+        let out2 = unsafe { std::mem::transmute_copy(&out) };
+        std::mem::forget(out);
+        out2
```

Comment:
`// TODO(mingwei)`
Not sure if there is a safe way to do this, but should add a safety comment/convince ourselves it is sane

Reply:
simply changed comment for now